EIPS/eip-FOCIL.md

```diff
@@ -40,12 +40,55 @@ TBD
-- **`Slot N`, `t=11s`**: The block proposer freezes its local inclusion lists view and IL committee members stop gossiping.
+- **`Slot N`, `t=9s`**: IL committee members freeze their view of a set of local ILs and no longer produce new ones.
+- **`Slot N`, `t=9 to 11s`**: IL committee members continue forwarding the local ILs they are aware of but ignore any new ones. The block proposer and attesters of `slot N+1` continue listening to gossiped local inclusion lists. To ensure none are omitted, the block proposer can request for any missing local inclusion lists from a specific IL committee member via an RPC endpoint, for example, at `t=10s`.
+- **`Slot N`, `t=11s`**: The block proposer freezes its local ILs view and IL committee members stop gossiping.
 - **`Slot N+1`, `t=0s`**: The block proposer broadcasts `block B` for `slot N+1` with an execution payload that satisfies the IL constraints.
 - **`Slot N+1`, `t=4s`**: The attesters accept `block B` only if it includes all transactions from the local inclusion lists, or if any missing transactions cannot be appended to the end of the execution payload, or if the block is full.
 
+#### Role and participants
+
+##### IL Committee Members
+
+- **`Slot N`, `t=0 to 8s`**:
+IL committee members construct their local ILs and broadcast them over the P2P network after processing the block for slot `N` and confirming it as the head. If no block is received by `t=7s`, they should run `get_head` and build and release their local ILs based on their node’s canonical head.
+
+By default, local ILs are built by selecting raw transactions from the public mempool, ordered by priority fees, up to the local IL’s maximum size in bits (e.g., 8 kb per local IL). Additional rules can be optionally applied to maximize CR, such as prioritizing valid transactions that have been pending in the mempool the longest.
+
+##### Nodes
+
+- **`Slot N`, `t=0 to 8s`**:
+Nodes receive local ILs from the P2P network and only forward and cache those that pass the CL P2P validation rules.
+
+- **`Slot N`, `t=9s`**:, IL freeze deadline:
+Nodes freeze their local ILs view, stop forwarding and caching new local ILs.
+
+---
+
+# CL P2P Validation Rules:
+
+1. The number of transactions in the local IL does not exceed the maximum gas limit allowed.
+2. The slot of the local IL matches the current slot. Local ILs not matching the current slot should be ignored.
+3. The parent hash of the IL is recognized.
+4. The IL is received before the local IL freeze deadline (e.g., 9s) into the slot.
+5. Received two or fewer local ILs from this IL committee member (see Local IL equivocation section below).
+6. The local IL is correctly signed by the validator.
+7. The validator is part of the IL committee.
+
+---
+
+##### Proposer
+
```

Comment:
I think it's good to define proposer's behavior between 0s and 11s. It has duty of listening to gossiped local ILs and caching them.

Reply:
agreed!